template.yaml

```diff
@@ -283,4 +289,74 @@ Resources:
       RestApiId: !Ref NvaFetchDoiFunctionApi
       StageName: !Ref NvaFetchDoiFunctionApi.Stage
       Timestamp: '${BUILD_TIMESTAMP}'
-      AuthorizerArn: !Ref CognitoAuthorizerArn
\ No newline at end of file
+      AuthorizerArn: !Ref CognitoAuthorizerArn
+
+  ScopusUnzipperFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: index.lambda_handler
+      MemorySize: !Ref LambdaMemorySize
+      Runtime: python3.8
+      Environment:
+        Variables:
+          XML_BUCKET_NAME: !Ref ScopusXmlBucketName
+      PackageType: Zip
+      InlineCode: |
+        import boto3, zipfile, os
+        from io import BytesIO
+        def lambda_handler(event, context):
+          XML_BUCKET = os.environ['XML_BUCKET_NAME']
+          ZIP_BUCKET = event['Records'][0]['s3']['bucket']['name']
+          ZIP_FILE = event['Records'][0]['s3']['object']['key']
+          client = boto3.client('s3')
+          resource=boto3.resource('s3')
+          zip_object = resource.Object(bucket_name=ZIP_BUCKET, key=ZIP_FILE)
+          buffer = BytesIO(zip_object.get()["Body"].read())
+          z = zipfile.ZipFile(buffer)
+          for filename in z.namelist():
+            response = client.put_object(
+              Body=z.open(filename).read(),
+              Bucket=ZIP_BUCKET if filename.endswith(".zip") else XML_BUCKET,
+              Key=f'unzipped/{filename}' if filename.endswith(".zip") else filename
+            )
```

Comment:
- Indent four spaces to comply with PEP8
- Use lowercase variable names (PEP8)
- What is happening with the "response" variable? Here, we just assign it multiple times without returning anything. The intention is the client.put_object, but we do nothing with the result.
  - Do we want to assess success across the processed files? (potentially retrying)
  - Do we want to report errors?